src/parser/view_transform.c

```diff
@@ -2755,6 +2755,7 @@ pt_pushable_query_in_pos (PARSER_CONTEXT * parser, PT_NODE * query, int pos)
   switch (query->node_type)
     {
     case PT_SELECT:
+    case PT_DBLINK_TABLE:
```

Comment:
Is PT_DBLINK_TABLE the same structure as PT_SELECT? 
Does the PT_DBLINK_TABLE node have query->info.query.q.select.list?

Reply:
info.query.q.select.list is null on the node PT_DBLINK_TABLE, but the node should be dealt as a pushable node in order to return "pushable = true"

Reply:
query->info.query.q.select.list may not be null.
PT_DBLINK_TABLE should be separated from existing routine.

Reply:
How about modifying as below:
```
case PT_DBLINK_TABLE:
  pushable = true;
  break;
```

Reply:
I can't comment because I haven't analyzed routines related to view. I'm analyzing the relevant routines as a whole, which I think will take quite some time.

Reply:
The logic to check pushable in the mq_copypush_sargable_terms function occupies a large part. However, I don't think the check is necessary in this case because the predicate is added to the dblink clause as shown below.
```
select * from ( subquery) where 'pushed predicate'
```
so real predicate push is performed at remote server.

I think adding dblink-related logic inside the mq_copypush_sargable_terms function will result in a lot of unintended behavior.
I would recommend completely separating the relevant logic from mq_copypush_sargable_terms ().

Reply:
mq_copypush_sargable_terms_dblink is added for separating routiine.

Reply:
This is a routine that examines the select list of a sub query.
this routine is not needed in dblink.
```
case PT_DBLINK_TABLE:
  pushable = true;
  break;
```

Reply:
Yes, the dblink is necessary to separate form the case.

Reply:
info is union. query->info.query.q.select.list may not be null.

Reply:
Ok, I separate the dblink from the case.